receiver/podmanreceiver/receiver.go

```diff
@@ -87,3 +170,53 @@ func (r *receiver) scrape(context.Context) (pdata.Metrics, error) {
 	}
 	return md, nil
 }
+
+func (r *receiver) handleEvents(ctx context.Context, eventBackoff *backoff.ExponentialBackOff) error {
+	c, err := r.clientFactory(r.set.Logger, r.config)
+	if err != nil {
+		return err
+	}
+	r.client = c
+
+	// Fetch the response from the endpoint
+	response, err := r.client.getEventsResponse(ctx)
+	if err != nil {
+		return err
+	}
+
+	dec := json.NewDecoder(response.Body)
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+			// Translate the response to the events format.
+			decodedEvent, err := decodeEvents(dec)
+			if err != nil {
+				if err == io.EOF {
+					break
```

Comment:
Why `break`? Does EOF means that we have to return from handleEvents? Or maybe just no new events, and we need to keep listening?

Reply:
Any kind of error means that it should return from handleEvents (which means the receiver will retry). So, agreed. `break` will not come here. Basically below code should not be here.
```
if err == io.EOF {
	break
}
```